tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/RabbitMQIntegration.cs

```diff
@@ -88,22 +89,23 @@ internal static string GetOperationName(Tracer tracer, string spanKind)
             };
         }
 
-        internal static void SetDataStreamsCheckpointOnProduce(Tracer tracer, Span span, RabbitMQTags tags, IDictionary<string, object> headers)
+        internal static void SetDataStreamsCheckpointOnProduce(Tracer tracer, Span span, RabbitMQTags tags, IBody body, IBasicProperties basicProperties)
         {
             var dataStreamsManager = tracer.TracerManager.DataStreamsManager;
-            if (dataStreamsManager == null || !dataStreamsManager.IsEnabled || headers == null)
+            if (dataStreamsManager == null || basicProperties == null || !dataStreamsManager.IsEnabled || basicProperties.Headers == null)
```

Comment:
as mentioned above, I'm not sure if basicProperties _can_ be null, but could potentially simplify to this anyway:
```suggestion
            if (dataStreamsManager == null || !dataStreamsManager.IsEnabled || basicProperties?.Headers == null)
```